arch/cortex-m4/src/lib.rs

```diff
@@ -66,30 +66,37 @@ pub unsafe extern "C" fn generic_isr() {
     mov r1, sp
     ldr r1, [r1, #4]
     stmia r1, {r4-r11}
+
+    /* Set thread mode to privileged */
+    mov r0, #0
+    msr CONTROL, r0
```

Comment:
`generic_isr` is only called to service interrupts, correct? My reading of B1.3.1 of the ARM v7m ARM is that the processor should be in Handler mode, which is by definition already privileged. I don't think this is needed.

Reply:
This is needed to make sure we're in privileged mode when we return the kernel, not for this handler's code.